### ISSUE TITLE
lead and lag API

### ISSUE THREAD

**Issue by GordStephen** [JuliaStats/TimeSeries.jl#221]
Right now `lead` and `lag` are defined over two methods each:

```julia
function lag{T,N}(ta::TimeArray{T,N}; period::Int=1)
function lag{T,N}(ta::TimeArray{T,N}, n::Int)
```

These could be consolidated nicely into a single method:
```julia
function lag{T,N}(ta::TimeArray{T,N}, n::Int=1)
```

which could then be called as either `lag(ta)` or `lag(ta, 2)`. But before I made that change, I figured I should check to see if the `period` keyword was a deliberate design decision (maybe to mirror another library's behaviour?) or just a means to an end that can be deprecated.

**Comment by milktrader** [comment]
R uses `k`, so us using `n` is acceptable. And agreed this is a nicer API. Is this a candidate for a deprecation warning? In any case, it certainly needs a mention in the NEWS under whatever minor release it gets tagged with. Also, the documentation will likely need an update.

**Comment by GordStephen** [comment]
Yep, definitely warrants a deprecation warning. This can be changed along with the padding option for lead/lag and introduced in 0.7.0, so in advance of that we can include a warning in the next 0.6.x release.

**Comment by GordStephen** [comment]
Implemented in #223
